posts/2021/10/interchange-dataframe-protocol-for-cudf.md

```diff
@@ -0,0 +1,341 @@
+All this work is done by a helper function `_from_dataframe` which builds up an entire cudf dataframe from an interchange dataframe object:
+
+```python
+from cudf.core.df_protocol import _from_dataframe
+df_rebuilt = _from_dataframe(dfo)
+print(f'rebuilt df\n----------\n{df_rebuilt}\n')
+print(f'df\n--\n{df}')
+```
+**output**:
+
+    rebuilt df
+    ----------
+        int  uint8 float   bool categorical
+    0  1000      0  <NA>   True        1000
+    1     2    128   2.5   <NA>           2
+    2   300    255  <NA>  False         300
+    3  <NA>     25  10.0   True        <NA>
+
+    df
+    --
+        int  uint8 float   bool categorical
+    0  1000      0  <NA>   True        1000
+    1     2    128   2.5   <NA>           2
+    2   300    255  <NA>  False         300
+    3  <NA>     25  10.0   True        <NA>
+
+
+We just went over a roundtrip demo from a cudf dataframe to the interchange dataframe object. Then we saw how to build a cudf dataframe object from the interchange dataframe object. Along the way, we've checked the integrity of the data.
+
+
+
+## Lessons learned on the go
+
+### Diversity advantage
+Many studies show the benefits and better performance of diverse teams. My experience in this project was the [`CONTRIBUTING.md`](https://github.com/rapidsai/cudf/blob/branch-21.08/CONTRIBUTING.md)(old version) document on the repository which was very unclear for me as a newcomer. Following my mentors' advice (@Ralf and @Kshiteej), I've opened an issue where I've shared my thoughts and kept asking clarifications which ended up in a (merged) [PR](https://github.com/rapidsai/cudf/pull/9026) to restructure the [`CONTRIBUTING.md`](https://github.com/rapidsai/cudf/blob/branch-21.12/CONTRIBUTING.md) (current version) document to make it clearer.
+
+Thus, a diversity of levels(experts, new comers, etc...) ensures an inclusive environment where everyone can find his way easily.
+
+### Test Driven Development (TDD)
+This process has been very helpful during this project. I've noticed my slowness on days where I've started developing features before writing any test. I kept going back and forth in the code base to ensure the coherence of different pieces of code written for that feature. However, when writing tests I felt the possibility to express all my expectations across different test cases then writing code for each one at a time. In this process, I felt like the tests pointed out next place on the code base where they might be something wrong. 
+
+### Collaboration, speaking out is very helpful
+Sometimes when stumble upon a problem, just speaking out or sharing the problem to someone else opens your eyes to a possible solution. This happens to me countless when discussing with my mentor @Kshiteej and my colleague @Alenka whose project is very close to mine. Otherwise, external inputs combined with ours will definitely be better than ours alone. 
```

Comment:
```suggestion
Sometimes, when stumbling upon a problem, just speaking out or sharing the problem to someone else opens your eyes to a possible solution. This happened to me countless times when discussing with my mentor @Kshiteej and my colleague @Alenka whose project is very close to mine. Otherwise, external inputs combined with ours will definitely be better than ours alone. 
```